### ISSUE TITLE
[MCU] Flash driver

### ISSUE THREAD

**Issue by AndreasFreund** [amosproj/amos2024ss07-updating-flash-boot-loader#11]
## User story
1. As a developer
2. I want / need to program the flash without dealing with hardware directly
3. So that my code stays portable

## Acceptance criteria
* .h-file without platform specific includes defines functions for (un)locking (if neccessary by this platform), erasing flash and writing to flash
* .h-file for platform has initialization method (if neccessary)
* .c-file implements initialization (if neccessary) and functionality

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
